cpp/src/ray/runtime/abstract_ray_runtime.cc

```diff
@@ -0,0 +1,112 @@
+
+#include "abstract_ray_runtime.h"
+
+#include <cassert>
+
+#include <ray/api.h>
+#include <ray/api/ray_config.h>
+#include <ray/api/ray_exception.h>
+#include "../util/address_helper.h"
+#include "local_mode_ray_runtime.h"
+
+namespace ray {
+namespace api {
+
+std::unique_ptr<AbstractRayRuntime> AbstractRayRuntime::ins_;
+std::once_flag AbstractRayRuntime::isInited_;
+
+AbstractRayRuntime &AbstractRayRuntime::DoInit(std::shared_ptr<RayConfig> config) {
```

Comment:
I don't understand why we need this abstract runtime. When would this ever be configurable? Shouldn't these methods just be calling into the core worker?